### PR TITLE
Routes : Added HTTP 405 handler middleware

### PR DESCRIPTION
**Pull request by nbcraft**
I added a 405 handler middleware to respect REST patterns :
http://www.restpatterns.org/HTTP_Status_Codes/405_-_Method_Not_Allowed

### CODE REVIEW

**Comment by nbcraft** [comment]
Hey :)
Did you have time to take a look at this ?
The Travis check seems to have failed while installing node 0.10
I don't feel like pushing a new commit just to trigger the checks again.
Thanks in advance ;)
Nathan

**Comment by nbcraft** [comment]
Bump :)

**Comment by aredridel** [comment]
I re-ran CI and it passed-- error in travis infrastructure.

File: lib/expressroutes.js [hunk 1]
```diff
@@ -132,6 +142,20 @@ function makeExpressRoute(router, mountpath, route) {
     router[route.method].apply(router, args);
 }
 
+/**
+ * Builds the middleware to manage not allowed calls that use wrong Method
+ * @param methods - list of avalaible method for this request
+ * @return {function}
+ */
+var buildNotAllowedMiddleware = function(methods) {
```
Comment (tlivings) [inline lib/expressroutes.js]: Code style in this module isn't to assign functions to variables.

File: lib/expressroutes.js [hunk 2]
```diff
@@ -151,9 +175,18 @@ function expressroutes(router, options) {
         res.json(options.api);
     });
 
+    var routesMethod = {};
```
Comment (tlivings) [inline lib/expressroutes.js]: Please hoist variables.

File: lib/expressroutes.js [hunk 3]
```diff
@@ -151,9 +175,18 @@ function expressroutes(router, options) {
...
+
     routes.forEach(function(route) {
         makeExpressRoute(router, mountpath, route);
+        var path = buildRoutePath(mountpath, route.path);
```
Comment (tlivings) [inline lib/expressroutes.js]: Hoist please.

File: lib/expressroutes.js [hunk 4]
```diff
@@ -151,9 +175,18 @@ function expressroutes(router, options) {
...
+        routesMethod[path] = routesMethod[path] || [];
+        routesMethod[path].push(route.method.toLowerCase());
     });
+
+    for(var path in routesMethod) {
```
Comment (tlivings) [inline lib/expressroutes.js]: Prefer `Object.keys...forEach` to `for...in`. Also white space after `for`, `if`, `function` (anonymous), etc...

**Comment by tlivings** [comment]
Other than the code style comments, I need to think about this approach a little more.

**Comment by nbcraft** [comment]
Penny for your thoughts? :wink:

**Comment by tlivings** [comment]
Yeah, I think this looks good to me. Also cc @subeeshcbabu.

**Pull request merged**